app/controllers/AiModelController.scala

```diff
@@ -0,0 +1,224 @@
+        dataStore <- dataStoreDAO.findOneByName(dataset._dataStore) ?~> "dataStore.notFound"
+        _ <- Fox
+          .serialCombined(request.body.trainingAnnotations.map(_.annotationId))(annotationDAO.findOne) ?~> "annotation.notFound"
+        modelId = ObjectId.generate
+        organization <- organizationDAO.findOne(request.identity._organization)
+        jobCommand = JobCommand.train_model
+        commandArgs = Json.obj(
+          "training_annotations" -> Json.toJson(trainingAnnotations),
+          "organization_name" -> organization.name,
+          "model_id" -> modelId,
+          "workflow_yaml" -> request.body.workflowYaml
+        )
+        newTrainingJob <- jobService
+          .submitJob(jobCommand, commandArgs, request.identity, dataStore.name) ?~> "job.couldNotRunTrainModel"
+        existingAiModelsCount <- aiModelDAO.countByNameAndOrganization(request.body.name,
+                                                                       request.identity._organization)
+        _ <- bool2Fox(existingAiModelsCount == 0) ?~> "model.nameInUse"
+        newAiModel = AiModel(
+          _id = modelId,
+          _organization = request.identity._organization,
+          _dataStore = dataStore.name,
+          _user = request.identity._id,
+          _trainingJob = Some(newTrainingJob._id),
+          _trainingAnnotations = trainingAnnotations.map(_.annotationId),
+          name = request.body.name,
+          comment = request.body.comment,
+          category = request.body.aiModelCategory
+        )
+        _ <- aiModelDAO.insertOne(newAiModel)
+        newAiModelJs <- aiModelService.publicWrites(newAiModel)
+      } yield Ok(newAiModelJs)
+  }
+
+  def runInference: Action[RunInferenceParameters] =
+    sil.SecuredAction.async(validateJson[RunInferenceParameters]) { implicit request =>
+      for {
+        _ <- userService.assertIsSuperUser(request.identity)
+        organization <- organizationDAO.findOne(request.identity._organization)
+        dataset <- datasetDAO.findOneByNameAndOrganization(request.body.datasetName, organization._id)
+        dataStore <- dataStoreDAO.findOneByName(dataset._dataStore) ?~> "dataStore.notFound"
+        // _ <- aiModelDAO.findOne(request.body.aiModelId) ?~> "aiModel.notFound"
+        _ <- datasetService.assertValidDatasetName(request.body.newDatasetName)
+        _ <- datasetService.assertNewDatasetName(request.body.newDatasetName, organization._id)
+        jobCommand = JobCommand.infer_with_model
+        boundingBox <- BoundingBox.fromLiteral(request.body.boundingBox).toFox
+        commandArgs = Json.obj(
+          "organization_name" -> organization.name,
+          "dataset_name" -> dataset.name,
+          "color_layer_name" -> request.body.coloLayerName,
+          "bounding_box" -> boundingBox.toLiteral,
+          "model_id" -> request.body.aiModelId,
+          "new_segmentation_layer_name" -> request.body.newSegmentationLayerName,
```

Comment:
I'd say we should also add a check to ensure `newSegmentationLayerName` follows the allowed characters for dataset layers and no `../../..` something